impl/src/planning.rs

```diff
@@ -416,82 +496,122 @@ impl<'planner> WorkspaceSubplanner<'planner> {
 
   /** Produces a crate context for each declared crate and dependency. */
   fn produce_crate_contexts(&self) -> Result<Vec<CrateContext>> {
+    // Build a list of all catalogs we want the context for
+    let mut catalogs = vec![self.crate_catalog];
+    for catalog in self.binary_dependencies.iter() {
+      catalogs.push(catalog);
+    }
+
+    // Build a list of all workspace files
+    let mut files: Vec<&CargoWorkspaceFiles> =
+      self.binary_deps_files
+      .iter()
+      .map(|(_key, val)| val)
+      .collect();
+    files.push(self.files);
+
     // Gather the checksums for all packages in the lockfile
     // which have them.
     //
     // Store the representation of the package as a tuple
     // of (name, version) -> checksum.
     let mut package_to_checksum = HashMap::new();
-    if let Some(lock_path) = self.files.lock_path_opt.as_ref() {
-      let lockfile = Lockfile::load(lock_path.as_path())?;
-      for package in lockfile.packages {
-        if let Some(checksum) = package.checksum {
-          package_to_checksum.insert(
-            (package.name.to_string(), package.version),
-            checksum.to_string(),
-          );
+    for workspace_files in files.iter() {
+      if let Some(lock_path) = workspace_files.lock_path_opt.as_ref() {
+        let lockfile = Lockfile::load(lock_path.as_path())?;
+        for package in lockfile.packages {
+          if let Some(checksum) = package.checksum {
+            package_to_checksum.insert(
+              (package.name.to_string(), package.version),
+              checksum.to_string(),
+            );
+          }
         }
       }
     }
 
-    self
-      .metadata
-      .resolve
-      .as_ref()
-      .ok_or_else(|| RazeError::Generic("Missing resolve graph".into()))?
-      .nodes
+    // Additionally, the binary dependencies need to have their checksums added as well
+    for (pkg, info) in self.settings.binaries.iter() {
```

Comment:
In contrast to comment below, I think this could do to be a bit more functional. I don't feel strongly about it though.

Reply:
I think I could use a more direct suggestion here. I struggled a bit writing this section 😅